cql3/type_json.cc

```diff
@@ -228,6 +230,18 @@ struct from_json_object_visitor {
     }
     bytes operator()(const boolean_type_impl& t) {
         if (!value.IsBool()) {
+            if (value.IsString()) {
+                std::string str(rjson::to_string_view(value));
+                boost::trim_all(str);
+                boost::to_lower(str);
```

Comment:
Do `boost::trim_all` and `boost::to_lower` work on string_views too? It would mean less copying - unless I'm mistaken, this code would create 3 separate strings. Ideally, we should trim the view (which is cheap), and then transform it to lower chars, creating the final string.

Reply:
I didn't manage to get it working with `std::string_view`. However according to boost's docs [boost::to_lower](https://www.boost.org/doc/libs/1_50_0/doc/html/boost/algorithm/to_lower.html) and [boost::trim_all](https://www.boost.org/doc/libs/1_54_0/doc/html/boost/algorithm/trim_all.html) modify string in place, so imo there is only one copy of the string.

Reply:
How can `boost::trim_all` possibly modify an instance of std::string in place? In the general case (let's disregard small string optimization for simplicity), it's a piece of allocated memory, and trimming some characters from the front would mean that you need to "pop_front" from a string, and I don't think std::string is capable of doing so.

As for to_lower - yeah, here the situation is easier and modifying in place probably takes place without a reallocation.

Reply:
Just curious, have you tried a sequence like that?
1. `std::string_view view = rjson::to_string_view(value));`
2. `boost::trim_all(view);`
3. `std::string str(view);`
4. `boost::to_lower(str);`
? I can imagine why boost would refuse calling to_lower on a string_view, but trimming is quite a natural thing to do on views.

Reply:
Following an internal discussion - indeed, it is possible to trim without reallocation, by shifting the characters to the front and then popping back. Terrible compared to std::string_view, but works :) So I withdraw my accusations of reduntant allocations :upside_down_face:

Reply:
I've tried at the beginning and boost cannot trim `std::string_view`